### PR TITLE
System columns

### PR DESCRIPTION
**Pull request by chiragsanghvi**
This PR makes changes in the testcases and code for system columns as mentioned in this issue https://github.com/informatics-isi-edu/chaise/issues/1258.

A lot of test-cases have been modified to take into account system columns and some have been removed.

This PR should only be merged once Ermrest PR https://github.com/informatics-isi-edu/ermrest/pull/165 for system columns is merged and ErmrestDataUtils PR https://github.com/informatics-isi-edu/ErmrestDataUtils/pull/11.

### CODE REVIEW

File: test/specs/reference/tests/14.pseudo_columns.js [hunk 1]
```diff
@@ -474,10 +474,10 @@ exports.execute = function (options) {
 
             describe('when visible-columns annotation is not present for the context, ', function () {
                 describe('PseudoColumn for key, ', function () {
-                    it('if key is simple and its contituent columns are part of simple foreign key, should not be added (instead it should apply the PseudoColumn for foreignkey logic.)', function(done) {
+                    it('if key is simple and its constituent columns are part of simple foreign key, should not be added (instead it should apply the PseudoColumn for foreignkey logic.)', function(done) {
```
Comment (RFSH) [inline test/specs/reference/tests/14.pseudo_columns.js]: Now that you have changed the schema, this test case and the one after are not testing what they're supposed to test. Just `xit` them and add a `TODO` for me please.

File: test/specs/comment/tests/01.comment.js [hunk 2]
```diff
@@ -26,12 +26,6 @@ exports.execute = function (options) {
             expect(table.comment).toBe(null);
         });
 
-        it('this key with a comment should have a non-null .comment property.', function () {
```
[the remark this answers is not in the document]
Comment (chiragsanghvi) [reply test/specs/comment/tests/01.comment.js]: Done.

File: test/specs/common/tests/03.table.js [hunk 3]
```diff
@@ -16,11 +16,11 @@ exports.execute = function(options) {
             });
 
             it('when keys have the same length, should return the one that is all integer/serial.', function () {
```
[the remark this answers is not in the document]
Comment (chiragsanghvi) [reply test/specs/common/tests/03.table.js]: Done.

File: test/specs/faceting/tests/01.faceting.js [hunk 4]
```diff
@@ -378,22 +377,6 @@ exports.execute = function (options) {
                 
             });
             
-            describe("regarding alternative tables for main table, ", function () {
```
[the remark this answers is not in the document]
Comment (chiragsanghvi) [reply test/specs/faceting/tests/01.faceting.js]: Done.

File: test/specs/reference/tests/08.alternative_tables.js [hunk 5]
```diff
@@ -1405,7 +1406,7 @@ exports.execute = function (options) {
                 expect(tuple.reference).toBeDefined();
                 expect(tuple.reference._table.name).toBe(baseTable2);
                 expect(tuple.reference.displayname.value).toBe(baseTable2);
-                expect(tuple.reference._location.path).toBe(schemaNameEncoded + ":" + baseTable2Encoded+ "/id=00001&value=12");
+                expect(tuple.reference._location.path).toStartWith(schemaNameEncoded + ":" + baseTable2Encoded);
```
[the remark this answers is not in the document]
Comment (chiragsanghvi) [reply test/specs/reference/tests/08.alternative_tables.js]: Done.

File: test/specs/reference/tests/08.alternative_tables.js [hunk 6]
```diff
@@ -2184,7 +2185,7 @@ exports.execute = function (options) {
...
```
[the remark this answers is not in the document]
Comment (chiragsanghvi) [reply test/specs/reference/tests/08.alternative_tables.js]: Done.

File: test/specs/reference/tests/14.pseudo_columns.js (hunk 1 above)
Comment (chiragsanghvi) [reply test/specs/reference/tests/14.pseudo_columns.js]: Done.

**Comment by RFSH** [comment]
I updated this branch with the latest changes. While doing so I found a bug related to all-outbound foriegnkeys.

In the list of visible-columns, we could have all-outbound source paths which we are supposed to get the value of it in the same request as the main entity. The logic here was faulty and was not attaching values for the source paths that are not ending up in the shortestkey column. This was happening because if the last column was different, then the generated hash for their name would have been different from the name that we're attaching to the foreignkey objects.

**Comment by RFSH** [comment]
With the increase of runtime, I ran some experiments in chaise and ermrestjs to just find where we should improve and whether we should be worried now about the time limit or not. You can find the summary in [here](https://github.com/informatics-isi-edu/ermrestjs/wiki/Stateful-Vs.-Stateless-Automatic-Tests)

**Pull request merged**